optimizely/config/polling_manager.go

```diff
@@ -139,6 +136,26 @@ func NewPollingProjectConfigManager(exeCtx utils.ExecutionCtx, sdkKey string) *P
 // GetConfig returns the project config
 func (cm *PollingProjectConfigManager) GetConfig() (optimizely.ProjectConfig, error) {
 	cm.configLock.RLock()
+
+	if cm.projectConfig != nil {
+		defer cm.configLock.RUnlock()
+		return cm.projectConfig, cm.err
+	}
+
+	cm.configLock.RUnlock()
+	startTs := time.Now()
+	for time.Now().Before(startTs.Add(defaultWaitConfig)) {
+		cm.configLock.RLock()
+		if cm.projectConfig != nil {
+			defer cm.configLock.RUnlock()
+			return cm.projectConfig, cm.err
+		}
+		cm.configLock.RUnlock()
+		cmLogger.Debug("Waiting for Polling Manager to be initialized")
+		time.Sleep(time.Second)
```

Comment:
I'm not very qualified to review this, but two things seem less than ideal:
- Lots of locking & unlocking
- Must wait until the next second boundary after config is fetched

Why can't we use 2 channels, and on one we write a value after 10 seconds signaling timeout, and on the other we write the config after it's fetched, then select on those 2 channels?